server/src/service-layer/controllers/BookingController.ts

```diff
@@ -1,12 +1,61 @@
 import { AvailableDatesRequestModel } from "service-layer/request-models/UserRequests"
 import { AvailableDatesResponse } from "service-layer/response-models/PaymentResponse"
 import { Timestamp } from "firebase-admin/firestore"
+
 import BookingDataService from "data-layer/services/BookingDataService"
 import BookingSlotService from "data-layer/services/BookingSlotsService"
-import { Controller, Post, Route, Security, SuccessResponse, Body } from "tsoa"
+// import { AllUserBookingsRequestBody } from "service-layer/request-models/BookingRequests"
+import { AllUserBookingSlotsResponse } from "service-layer/response-models/BookingResponse"
+import {
+  Controller,
+  Get,
+  Post,
+  Route,
+  Security,
+  SuccessResponse,
+  Body,
+  Query
+} from "tsoa"
+import { firestoreTimestampToDate } from "data-layer/adapters/DateUtils"
 
 @Route("bookings")
 export class BookingController extends Controller {
+  @SuccessResponse("200", "Found bookings")
+  @Security("jwt", ["member"])
+  @Get()
+  public async getAllBookings(
+    @Query() uid: string
```

Comment:
the problem with including the UID like this is that members could technically query the bookings of others if they know the `uid`. The safer way to do it is extract their `uid` from the user object returned in the request (which we do in the `Authentication.ts` middleware). 

have a look at the `getSelf()` method in `server/src/service-layer/controllers/UserController.ts`